### PR TITLE
VEN-670 | Delete unused icons and check the used ones if they are replaceable with HDS icons

### PR DESCRIPTION
**Pull request by TatuArvela**
## Description :sparkles:

* Deleted unused icon files and icons that can be replaced with HDS icons
* Replaced icon styles with HDS icon styles to allow interchangeable use without `Icon`
  * `Trash` icon design is a little different
  * All icons now follow the color of the component they are in (allows easy disabled coloring), unless otherwise specified with a style
* Replaced `Icon` with a derived `IconWrapper` component which allows using the explicitly specified colors and outline as before, and also supports HDS icons
* Update `hds-react `to the latest version (0.11.3) to have more icons available
  * This brings minor changes to element sizing, and also adds a required * token to inputs

### Closes :no_good_woman:

* [VEN-670](https://helsinkisolutionoffice.atlassian.net/browse/VEN-670): Delete unused icons and check the used ones if they are replaceable with HDS icons

### Automated tests :gear:️

* Related snapshots have been updated

### Manual testing :construction_worker_man:

* All pages and views have to be checked manually

### CODE REVIEW

File: src/common/iconWrapper/IconWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  size?: 'xs' | 's' | 'm' | 'l' | 'xl';
+}
+
+const IconWrapper: React.FC<IconWrapperProps> = ({ className, icon, outlined, color = 'standard', size = 's' }) => {
+  return (
+    <div
+      className={classNames(
+        styles.iconWrapper,
+        styles[color],
+        styles[size],
```
[the remark this answers is not in the document]
Comment (TatuArvela) [reply src/common/iconWrapper/IconWrapper.tsx:21]: Correct, this seems like a leftover from the old code. I'll fix it

File: src/common/icons/index.ts [hunk 2]
```diff
@@ -1,41 +1,23 @@
...
-export { default as IconArrowLeft } from './IconArrowLeft';
-export { default as IconArrowRight } from './IconArrowRight';
-export { default as IconCalendar } from './IconCalendar';
-export { default as IconCalendarAdd } from './IconCalendarAdd';
-export { default as IconDirections } from './IconDirections';
-export { default as IconHouse } from './IconHouse';
-export { default as IconPerson } from './IconPerson';
-export { default as IconRead } from './IconRead';
-export { default as IconAccessibility } from './IconAccessibility';
+export type { IconProps } from './Icon.interface';
```
[the remark this answers is not in the document]
Comment (TatuArvela) [reply src/common/icons/index.ts:1]: Either way goes for me, what do others think?
Currently the icons use IconProps from `Icon.interface`, while IconWrapper uses it from `index.ts`.

File: src/common/iconWrapper/IconWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import styles from './iconWrapper.module.scss';
+import { IconProps } from '../icons';
+
+export interface IconWrapperProps {
+  className?: string;
+  outlined?: boolean;
+  icon: React.FC<IconProps>;
```
[the remark this answers is not in the document]
Comment (TatuArvela) [reply src/common/iconWrapper/IconWrapper.tsx]: Fixed this one as well

File: src/common/berthDetails/BerthDetails.tsx [hunk 4]
```diff
@@ -74,52 +75,47 @@ const BerthDetails: React.SFC<BerthDetailsProps> = ({
...
+              className={styles.property}
+              icon={IconTrash}
+              label={t('offer.berthDetails.waste')}
+              active={wasteCollection}
+            />
           )}
           {isDefined(isAccessible) && (
-            <div className={styles.property}>
-              <Icon shape="IconAccessibility" color={getColor(isAccessible)} outlined />
-              <Text className={styles.propertyLabel} color={getColor(isAccessible)}>
```
Comment (Nizar-Rahme) [inline src/common/berthDetails/BerthDetails.tsx:119]: I assume you forgot to delete the no-longer-needed styles for `propertyLabel`.

File: src/common/iconWrapper/IconWrapper.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+import styles from './iconWrapper.module.scss';
+import { IconProps } from '../icons';
+
+export interface IconWrapperProps {
+  className?: string;
+  outlined?: boolean;
+  icon: (props: IconProps) => React.ReactElement | null;
+  color?: 'standard' | 'disabled' | 'brand' | 'secondary' | 'critical';
+  size?: 'xs' | 's' | 'm' | 'l' | 'xl';
```
Comment (Nizar-Rahme) [inline src/common/iconWrapper/IconWrapper.tsx:12]:
```TSX
import { IconProps as HDSIconProps } from 'hds-react';
```
```suggestion
  size: HDSIconProps['size'];
```

File: src/common/icons/Icon.interface.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import { SVGAttributes } from 'react';
+
+export type IconProps = SVGAttributes<SVGElement> & {
+  /**
+   * Icon size
+   */
+  size?: 'xs' | 's' | 'm' | 'l' | 'xl';
```
Comment (Nizar-Rahme) [inline src/common/icons/Icon.interface.ts:7]: And here, if we can get these types directly from HDS.

**Comment by TatuArvela** [comment]
> All `fill-rule` attributes need to be changed to `fillRule`

Fixed

**Pull request merged**